python/tritonserver/_api/_response.py

```diff
@@ -360,13 +104,12 @@ class InferenceResponse:
     @staticmethod
     def _from_tritonserver_inference_response(
         model: _model.Model,
-        request: TRITONSERVER_InferenceRequest,
         response,
         flags: TRITONSERVER_ResponseCompleteFlag,
+        output_memory_type: Optional[DeviceOrMemoryType] = None,
     ):
         result = InferenceResponse(
             model,
-            request.id,
```

Comment:
question: why omit this / change this - the idea here is that in case there is a fundamental error (i.e. an empty response, or some other error) - we get at least the request.id as stored in the original request to tie this response to the error - as in line 118 we aren't guranteed a full response ...

Reply:
there were some issue with the request lifecycle, but it is fixed later, so it is safe to add by the request.id setting. updated: [Update inference response conversion interface](https://github.com/triton-inference-server/core/pull/426/commits/023a1459d84d26b42a80b659de2a5882cb62a4f8)